wormhole/file_transport.go

```diff
@@ -449,23 +490,46 @@ func (t *fileTransport) listen() error {
 	if testDisableLocalListener {
 		return nil
 	}
+	if t.relayProto == "" || t.relayProto == "tcp" {
+		l, err := net.Listen("tcp", ":0")
+		if err != nil {
+			return err
+		}
 
-	l, err := net.Listen("tcp", ":0")
-	if err != nil {
-		return err
+		t.listener = l
+	}
+
+	if t.relayProto == "ws" {
+		t.listener = nil
 	}
 
-	t.listener = l
 	return nil
 }
 
 func (t *fileTransport) listenRelay() error {
+
+	ctx := context.Background()
 	if t.relayAddr == "" {
 		return nil
 	}
-	conn, err := net.Dial("tcp", t.relayAddr)
-	if err != nil {
-		return err
+	var conn net.Conn
+	var err error
+	if t.relayProto == "tcp" {
+		conn, err = net.Dial("tcp", t.relayAddr)
+		if err != nil {
+			return err
+		}
+	}
+
+	if t.relayProto == "ws" {
+		// TODO: The hardcoding of the URL should be removed once there is
+		// a way to represent it in the Hint messages. At the moment, there
+		// is no way to do that and hence this hardcoding.
+		c, _, err := websocket.Dial(ctx, "ws://" + t.relayAddr + "/v1", nil)
```

Comment:
I would do something like this:
```golang
addr := fmt.Sprintf("ws://%s/v1", t.relayAddr)
c, _, err := websocket.Dial(ctx, addr, nil)
```

In the event that `t.relayAddr` changes to a struct, this should work as long as it implements a `String` method. I'm pretty sure the addition wouldn't be defined in that case.